ssl/record/rec_layer_s3.c

```diff
@@ -853,15 +865,19 @@ int do_ssl3_write(SSL *s, int type, const unsigned char *buf,
         if (s->compress != NULL)
             maxcomplen += SSL3_RT_MAX_COMPRESSED_OVERHEAD;
 
-        /* write the header */
-        if (!WPACKET_put_bytes_u8(thispkt, rectype)
+        /*
+         * When using offload kernel will write the header.
+         * Otherwise write the header now
+         */
+        if (!BIO_get_ktls(s->wbio)
+                && (!WPACKET_put_bytes_u8(thispkt, rectype)
                 || !WPACKET_put_bytes_u16(thispkt, version)
```

Comment:
Expanded the comment.
The version was set when kTLS was started:
`crypto_info.info.version = s->version;`

I think this code might need to be updated once ktls support is expanded to TLS1.3,. However, we will try to avoid it.